### PR TITLE
Document alternatives

### LINKED ISSUE #400: Document pros/cons to alternative tools
Followup of https://twitter.com/julien_lafont/status/921691101198774277

It would be good to compare scalafix to wartremover/scapegoat/scalastyle/abide in our docs.

### PR DESCRIPTION
**Pull request by olafurpg**
Fixes #400

### CODE REVIEW

File: website/src/main/tut/docs/users/faq.md [hunk 1]
```diff
@@ -10,6 +10,115 @@ If you have any questions, don't hesitate to ask on {% gitter %}.
...
+
+[Scalastyle](http://www.scalastyle.org/) is a Scala style checker.
+
+* Scalastyle has a lot more rules than Scalafix out of the box.
+* Scalafix supports semantic and syntactic rules while Scalastyle supports only syntactic rules.
+  This means that Scalafix rules know more about the types/symbols in your source code.
+  For example, Scalafix rules can tell if `get` in `x.get` comes from `Option[T].get` or from another class that happens to also have a `.get` method.
+  Scalastyle is purely syntactic, which means it does not support rules that need information about types/symbols.
+* Scalastyle runs as a separate tool outside of compilation, just like Scalafix.
+* Scalafix rules uses the [Scalameta](http://scalameta.org/) AST while Scalastyle rules use
```
Comment (ShaneDelmore) [inline website/src/main/tut/docs/users/faq.md]: While Scalastyle rules use....?

File: website/src/main/tut/docs/users/faq.md [hunk 2]
```diff
@@ -10,6 +10,136 @@ If you have any questions, don't hesitate to ask on {% gitter %}.
...
+
+* Scalafix uses the Scala 2.x compiler to resolve symbols/types while
+  the IntelliJ Scala plugin uses it's own own typechecker.
+  This means that if a project compiles with the Scala compiler,
+  then scalafix can analyze the source code (assuming no conflicting compiler plugins).
+  The IntelliJ Scala typechecker is known to produce false red squigglies.
+  It depends on what Scala features you use whether the IntelliJ Scala can analyze
+  your source code.
+* Scalafix is primarily aimed to be used in batch-mode through a console interface
+  while IntelliJ is primarily aimed for interactive use in the IntelliJ IDE.
```
Comment (laughedelic) [inline website/src/main/tut/docs/users/faq.md]: Can Scalafix be used to integrate with an LSP editor-client linting/code-actions?

Comment (olafurpg) [reply website/src/main/tut/docs/users/faq.md]: Yes it can :) https://github.com/scalameta/language-server/pull/1

Comment (laughedelic) [reply website/src/main/tut/docs/users/faq.md]: Oh my!!! I was just thinking exactly about this project last days! Great 👍 👍 👍  I'm really glad you are already working on it. I'm going to follow development closely.

**Pull request merged**